packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -2635,10 +2635,12 @@ function mountMemo<T>(
 ): T {
   const hook = mountWorkInProgressHook();
   const nextDeps = deps === undefined ? null : deps;
+  const nextValue = nextCreate();
   if (shouldDoubleInvokeUserFnsInHooksDEV) {
+    setIsStrictModeForDevtools(true);
     nextCreate();
+    setIsStrictModeForDevtools(false);
```

Comment:
Do we need to restore this in a `finally` block instead?
```suggestion
    setIsStrictModeForDevtools(true);
    try {
      nextCreate();
    } finally {
      setIsStrictModeForDevtools(false);
    }
```

Reply:
I considered this but it's not clear if this is necessary as functions passed to useMemo should not throw. Elsewhere, the render function _can_ throw so it makes sense to wrap it in a try-finally.